src/modules/accounting/staking_vaults.py

```diff
@@ -0,0 +1,669 @@
+                        "treeIndex": item["treeIndex"],
+                    }
+                )
+            return out
+
+        values = stringify_values(tree.values)
+
+        extra_values = {}
+        for vault_adr, vault_info in vaults.items():
+            extra_values[vault_adr] = {
+                "inOutDelta": str(vault_info.in_out_delta),
+                "prevFee":  str(vaults_fee_map[vault_adr].prev_fee),
+                "infraFee": str(vaults_fee_map[vault_adr].infra_fee),
+                "liquidityFee": str(vaults_fee_map[vault_adr].liquidity_fee),
+                "reservationFee": str(vaults_fee_map[vault_adr].reservation_fee),
+            }
+
+        output: dict[str, Any] = {
+            **dict(tree.dump()),
+            "refSlot": bs.ref_slot,
+            "blockHash": bs.block_hash,
+            "blockNumber": bs.block_number,
+            "timestamp": chain_config.genesis_time + bs.slot_number * chain_config.seconds_per_slot,
+            "extraValues": extra_values,
+            "prevTreeCID": prev_tree_cid,
+            "leafIndexToData": {
+                "0": "vaultAddress",
+                "1": "totalValueWei",
+                "2": "fee",
+                "3": "liabilityShares",
+                "4": "slashingReserve",
+            },
+        }
+        output.update(values=values)
+
+        dumped_tree_str = json.dumps(output, default=encoder)
+
+        cid = self.ipfs_client.publish(dumped_tree_str.encode('utf-8'), 'merkle_tree.json')
+
+        return cid
+
+    def get_ipfs_report(self, ipfs_report_cid: str) -> MerkleTreeData:
+        if ipfs_report_cid == "":
+            raise ValueError("Arg ipfs_report_cid could not be ''")
+        return self.get_vault_report(ipfs_report_cid)
+
+    def get_prev_cid(self, bs: BlockStamp) -> str:
+        report = self.lazy_oracle.get_report(block_identifier=bs.block_hash)
+        if report is None:
+            return ""
+        return report.cid
+
+    def _calculate_pending_deposits_balances(
+        self,
+        pending_deposits: list[PendingDeposit],
+        vault_validators: list[Validator],
+        vault_pending_deposits: list[PendingDeposit],
+        vault_withdrawal_credentials: str,
+    ) -> int:
+        vault_validator_pubkeys = set(validator.validator.pubkey for validator in vault_validators)
+        deposits_by_pubkey: dict[str, list[PendingDeposit]] = defaultdict(list)
+
+        for deposit in vault_pending_deposits:
+            deposits_by_pubkey[deposit.pubkey].append(deposit)
+
+        total_value = 0
+
+        for pubkey, deposits in deposits_by_pubkey.items():
+            deposit_value = sum(Web3.to_wei(int(deposit.amount), 'gwei') for deposit in deposits)
+
+            # Case 1: Validator exists and is already bound to this vault
+            if pubkey in vault_validator_pubkeys:
+                total_value += deposit_value
+                continue
+
```

Comment:
:warning: :warning: we should still have `continue` case when the pubkey belong to the non-vaults validator set